### PR TITLE
Bug 1107139 - Show id-overlay when there's no screenshot

### PR DESCRIPTION
**Pull request by sfoster**
WIP, show the notification-overlay when there's no _screenshotBlob

### CODE REVIEW

File: apps/system/style/window.css [hunk 1]
```diff
@@ -337,9 +337,15 @@
...
+  opacity: 1;
+}
+
+
 /* we don't want to show the overlay at all during the opening
    transition (from the homescreen and from the cardview) */
-#screen .appWindow.in-task-manager.overlay > .identification-overlay,
+#screen .appWindow.in-task-manager:not(.no-screenshot).overlay > .identification-overlay,
 .appWindow.enlarge.overlay > .identification-overlay,
 .appWindow.from-cardview.overlay > .identification-overlay {
```
Comment (etiennesegonzac) [inline apps/system/style/window.css]: adding a `:not(.no-screenshot)` here should help for the transition back to the app

File: apps/system/js/app_window.js [hunk 2]
```diff
@@ -1294,9 +1297,15 @@
...
       var screenshotURL = this.requestScreenshotURL();
-      this.screenshotOverlay.style.backgroundImage =
-        'url(' + screenshotURL + ')';
+      this.screenshotOverlay.style.backgroundImage = screenshotURL ?
+          'url(' + screenshotURL + ')' : 'none';
+      if (screenshotURL) {
+        this.element.classList.remove('no-screenshot');
+      } else {
+        this.element.classList.add('no-screenshot');
+      }
```
Comment (etiennesegonzac) [inline apps/system/js/app_window.js]: this.element.classList.toggle('no-screenshot', !screenshotURL)

File: apps/system/style/window.css [hunk 3]
```diff
@@ -337,9 +337,15 @@
   opacity: 1;
 }
 
+.appWindow.in-task-manager.overlay.no-screenshot > .identification-overlay {
+  visibility: visible;
+  opacity: 1;
```
Comment (etiennesegonzac) [inline apps/system/style/window.css]: maybe we should add `transition: opacity 0.2s ease-in, visibility 0.2s;` to transition both ways in this case

**Comment by etiennesegonzac** [comment]
A last issue I have (testing with the marketplace app) is that we also try to get a screenshot in `handle__closed` so sometimes the overlay appears properly but then the app gets the `_closed` event, requests a new screenshot, gets it, then flashes badly.

I think we should early return in `handle__closed` if the app isn't loaded (`!this.loaded`) to prevent a crappy screenshot from a partially loaded app to screw with the clean identification overlay.

File: apps/system/style/window.css (hunk 3 above)
Comment (etiennesegonzac) [inline apps/system/style/window.css]: as opposed to the rest of the time where we make the overlay appear with no transition and only transition if we're removing it.

**Comment by try-server-hook** [comment]
[sfoster Sam Foster (sfoster)](https://github.com/sfoster) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=bde629ac97aa)

**Comment by try-server-hook** [comment]
[sfoster Sam Foster (sfoster)](https://github.com/sfoster) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=d5313ef5a3ba)

File: apps/system/test/marionette/task_manager_test.js [hunk 4]
```diff
@@ -125,6 +128,55 @@ marionette('Task Manager', function() {
...
+          });
+          return req;
+        };
+        // reset screenshotBlob state
+        var app = win.Service.currentApp;
+        app._screenshotBlob = null;
+      }, [iframeId]);
+
+      taskManager.show();
+    });
```
Comment (etiennesegonzac) [inline apps/system/test/marionette/task_manager_test.js]: nit: empty line here

File: apps/system/test/marionette/task_manager_test.js [hunk 5]
```diff
@@ -125,6 +128,55 @@ marionette('Task Manager', function() {
...
+      var screenshot = card.findElement(taskManager.selectors.screenshot);
+      var backgroundImage = screenshot.cssProperty('background-image');
+      assert(backgroundImage.indexOf('blob') == -1);
+
+      var instanceId = card.getAttribute('data-app-instance-id');
+      var appElement = client.findElement('#' + instanceId);
+      var idOverlay =  appElement.findElement('.identification-overlay');
+      assert(idOverlay);
+      assert(idOverlay.displayed());
+
```
Comment (etiennesegonzac) [inline apps/system/test/marionette/task_manager_test.js]: nit: and no there

File: apps/system/test/marionette/task_manager_test.js [hunk 6]
```diff
@@ -125,6 +128,55 @@ marionette('Task Manager', function() {
...
 
+  suite('when launched while an app is still initializing', function() {
+    var slowApp;
+    setup(function() {
+      // launch a 3rd app which we can mock easily
+      slowApp = new FakeApp(client, 'app://' + slowAppOrigin);
+      slowApp.launch();
+
+      var iframeId = slowApp.iframe.getAttribute('id');
+      // mock iframe.getScreenshot so the screenshot never appears
```
Comment (etiennesegonzac) [inline apps/system/test/marionette/task_manager_test.js]: not a big fan of mocking in integration tests but I don't see how to test this without triggering nightmarish intermittents without it :)

**Comment by try-server-hook** [comment]
[sfoster Sam Foster (sfoster)](https://github.com/sfoster) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4b936f9c034e)

**Comment by try-server-hook** [comment]
[sfoster Sam Foster (sfoster)](https://github.com/sfoster) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=aedfd7df184e)

**Pull request merged**